File: engine/effects/handlers/summon.py
"""Summon effect handler scaffold."""

from __future__ import annotations

from engine.effects.base import Effect


def handle_summon(effect: Effect) -> dict:
    return {
        "status": "ok",
        "kind": "summon",
        "source": effect.source_unit_id,
        "payload": effect.payload,
    }

File: src/effects/handlers/summon.ts
/**
 * Summon effect handler scaffold.
 * Mirrors engine/effects/handlers/summon.py
 */

import { Effect } from "../base";

export function handleSummon(effect: Effect): Record<string, unknown> {
  return {
    status: "ok",
    kind: "summon",
    source: effect.sourceUnitId,
    payload: effect.payload,
  };
}
